scipy/ndimage/_filters.py

```diff
@@ -1318,20 +1318,45 @@ def maximum_filter(input, size=None, footprint=None, output=None,
 
 @_ni_docstrings.docfiller
 def _rank_filter(input, rank, size=None, footprint=None, output=None,
-                 mode="reflect", cval=0.0, origin=0, operation='rank'):
+                 mode="reflect", cval=0.0, origin=0, operation='rank',
+                 axes=None):
     if (size is not None) and (footprint is not None):
         warnings.warn("ignoring size because footprint is set", UserWarning, stacklevel=3)
     input = numpy.asarray(input)
     if numpy.iscomplexobj(input):
         raise TypeError('Complex type not supported')
-    origins = _ni_support._normalize_sequence(origin, input.ndim)
+    axes = _ni_support._check_axes(axes, input.ndim)
+    num_axes = len(axes)
+    origins = _ni_support._normalize_sequence(origin, num_axes)
     if footprint is None:
         if size is None:
             raise RuntimeError("no footprint or filter size provided")
-        sizes = _ni_support._normalize_sequence(size, input.ndim)
+        sizes = _ni_support._normalize_sequence(size, num_axes)
         footprint = numpy.ones(sizes, dtype=bool)
     else:
         footprint = numpy.asarray(footprint, dtype=bool)
+    if num_axes < input.ndim:
+        # set origin = 0 for any axes not being filtered
+        origins_temp = [0,] * input.ndim
+        for o, ax in zip(origins, axes):
+            origins_temp[ax] = o
+        origins = origins_temp
+
+        if not isinstance(mode, str) and isinstance(mode, Iterable):
+            # set mode = 'constant' for any axes not being filtered
+            modes = _ni_support._normalize_sequence(mode, num_axes)
+            modes_temp = ['constant'] * input.ndim
+            for m, ax in zip(modes, axes):
+                modes_temp[ax] = m
+            mode = modes_temp
+
+        # insert singleton dimension along any non-filtered axes
+        if footprint.ndim != num_axes:
+            raise RuntimeError("footprint.ndim must match len(axes)")
```

Comment:
Looks like this needs a test. Also, I would have thought this would be a `ValueError`.

Reply:
There are many other pre-existing checks in `_filters.py` that already use `RuntimeError` when the window or footprint shape is wrong, so I used the same here for consistency. Agree that `ValueError` is perhaps a better choice, but not sure we want to change it everywhere at this point?

Reply:
Ok. Keep it consistent.